### DISCUSSION TITLE
I can't get Task Dependencies to work

### DISCUSSION THREAD

**Discussion by leo-petrucci** [vercel/turborepo#5256] [Help]
I'm a bit confused, the docs page make it seem very simple but I can't get Turbo to play along.

According to the [Arbitrary Worskpaces example](https://turbo.build/repo/docs/core-concepts/monorepos/task-dependencies#from-arbitrary-workspaces), I should just be able to specify `<workspace>#<task>` in `turbo.json` and have Turbo launch both tasks.

```
    "dev": {
      "dependsOn": ["frontend#dev", "api#dev"]
    },
```

When I run Turbo however this is the output:

```
> turbo dev
yarn table
 ERROR  run failed: error preparing engine: Could not find "api#dev" in root turbo.json or "api" workspace
Turbo error: error preparing engine: Could not find "api#dev" in root turbo.json or "api" workspace
```

The workspace does exist and both commands run fine on their own:

```
> turbo frontend#dev
• Remote caching disabled
frontend:dev cache miss, executing 6e2e35dbb79fbce9
```

I can only get the command to work if I modify it as such:

```
    "dev": {
      "dependsOn": ["frontend#dev", "api#dev"]
    },
    "frontend#dev": {},
    "api#dev": {},
```

But this seems excessively verbose and considering the amount of projects in my repo I'd end up having a lot of empty Turbo configs.

Am I misunderstanding how this works?

I'm using `turbo@1.10.2`.

**Comment by mehulkar** [comment]
Can you provide a minimal repro? The example you give above should actually cause a circular dependency validation error (since you're askind all `dev` tasks to depend on specific `dev` tasks). I tried this out locally with `turbo@1.10.2` and saw the circular dependency error as I expected, so I feel like there may be something else missing from your question.

**Comment by anthonyshew** [comment]
Going to close here for staleness.
